src/json_get_array.rs

```diff
@@ -0,0 +1,92 @@
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow::array::ListArray;
+use arrow_schema::{DataType, Field};
+use datafusion_common::arrow::array::ArrayRef;
+use datafusion_common::{Result as DataFusionResult, ScalarValue};
+use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
+use jiter::Peek;
+
+use crate::common::{check_args, get_err, invoke, jiter_json_find, GetError, JsonPath};
+use crate::common_macros::make_udf_function;
+use crate::common_union::JsonUnion;
+
+make_udf_function!(
+    JsonGetArray,
+    json_get_array,
+    json_data path,
+    r#"Get an arrow array value from a JSON string by its "path""#
+);
+
+#[derive(Debug)]
+pub(super) struct JsonGetArray {
+    signature: Signature,
+    aliases: [String; 1],
+}
+
+impl Default for JsonGetArray {
+    fn default() -> Self {
+        Self {
+            signature: Signature::variadic_any(Volatility::Immutable),
+            aliases: ["json_get_array".to_string()],
+        }
+    }
+}
+
+impl ScalarUDFImpl for JsonGetArray {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        self.aliases[0].as_str()
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, arg_types: &[DataType]) -> DataFusionResult<DataType> {
+        check_args(arg_types, self.name()).map(|()| DataType::List(Field::new_list_field(DataType::Utf8, true).into()))
+    }
+
+    fn invoke(&self, args: &[ColumnarValue]) -> DataFusionResult<ColumnarValue> {
+        let to_array = |c: JsonUnion| {
+            let array: ListArray = c.try_into()?;
+            Ok(Arc::new(array) as ArrayRef)
+        };
+        invoke::<JsonUnion, String>(args, jiter_json_get_array, to_array, ScalarValue::Utf8)
+    }
+
+    fn aliases(&self) -> &[String] {
+        &self.aliases
+    }
+}
+
+fn jiter_json_get_array(json_data: Option<&str>, path: &[JsonPath]) -> Result<String, GetError> {
```

Comment:
I think you want to build up and return a `UnionArray` here, not return a string.